controller/tap/apiserver.go

```diff
@@ -56,39 +85,48 @@ func NewAPIServer(
 		log:            log,
 	}
 
-	router := initRouter(h)
+	lis, err := net.Listen("tcp", addr)
+	if err != nil {
+		return nil, fmt.Errorf("net.Listen failed with: %s", err)
+	}
 
-	server := &apiServer{
-		router:       router,
+	s := &APIServer{
+		Server:       httpServer,
+		listener:     lis,
+		router:       initRouter(h),
 		allowedNames: allowedNames,
+		certValue:    &emptyCert,
 		log:          log,
 	}
+	s.Handler = prometheus.WithTelemetry(s)
+	httpServer.TLSConfig.GetCertificate = s.getCertificate
 
-	clientCertPool := x509.NewCertPool()
-	clientCertPool.AppendCertsFromPEM([]byte(clientCAPem))
+	if err := watcher.UpdateCert(s.certValue); err != nil {
+		return nil, fmt.Errorf("Failed to initialized certificate: %s", err)
+	}
 
-	wrappedServer := prometheus.WithTelemetry(server)
+	go watcher.ProcessEvents(log, s.certValue, updateEvent, errEvent)
 
-	s := &http.Server{
-		Addr:    addr,
-		Handler: wrappedServer,
-		TLSConfig: &tls.Config{
-			Certificates: []tls.Certificate{cert},
-			ClientAuth:   tls.VerifyClientCertIfGiven,
-			ClientCAs:    clientCertPool,
-		},
-	}
+	return s, nil
+}
 
-	lis, err := net.Listen("tcp", addr)
-	if err != nil {
-		log.Fatalf("net.Listen failed with: %s", err)
+// Start starts the https server
```

Comment:
```suggestion
// Start the https server
```

Reply:
The first is the name of the function and the second the verb describing what it does. Although it seems like repetition, I think the original comment is correct.

Reply:
Oh right yea with the repetition it reads a little weird but I agree it's correct as is.